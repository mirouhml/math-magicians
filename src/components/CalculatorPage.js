import Calculator from './Calculator';

const CalculatorPage = () => (
  <div className="background calculator-page">
    <div>
      <h2>Let&apos;s do some math!</h2>
    </div>
    <div className="calculator">
      <Calculator />
    </div>
  </div>
);

export default CalculatorPage;
