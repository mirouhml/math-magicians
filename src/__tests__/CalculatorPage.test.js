import { screen, render } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import CalculatorPage from '../components/CalculatorPage';

describe('CalculatorPage component tests:', () => {
  it('CalculatorPage component renders correctly', () => {
    const tree = renderer
      .create(<CalculatorPage />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Check if the title of the page is rendered', () => {
    render(<CalculatorPage />);
    const title = screen.getByText('Let\'s do some math!');
    expect(title).toBeInTheDocument();
  });

  it('Check if the calculator is rendered', () => {
    const { container } = render(<CalculatorPage />);
    const calculator = container.querySelector('.calculator-table');
    expect(calculator).toBeInTheDocument();
  });
});
