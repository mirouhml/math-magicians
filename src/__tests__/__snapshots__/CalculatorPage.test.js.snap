// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalculatorPage component tests: CalculatorPage component renders correctly 1`] = `
<div
  className="background calculator-page"
>
  <div>
    <h2>
      Let's do some math!
    </h2>
    <img
      alt="Math Gif"
      src="https://upload.wikimedia.org/wikipedia/commons/e/e2/Math_-_Idil_Keysan_-_Wikimedia_Giphy_stickers_2019.gif"
    />
  </div>
  <div
    className="calculator"
  >
    <table
      className="calculator-table"
    >
      <tbody>
        <tr>
          <td
            className="result"
            colSpan="4"
          >
            0
          </td>
        </tr>
        <tr>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              AC
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              +/-
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              %
            </button>
          </td>
          <td>
            <button
              className="operator"
              onClick={[Function]}
              type="button"
            >
              รท
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              7
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              8
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              9
            </button>
          </td>
          <td>
            <button
              className="operator"
              onClick={[Function]}
              type="button"
            >
              ร
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              4
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              5
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              6
            </button>
          </td>
          <td>
            <button
              className="operator"
              onClick={[Function]}
              type="button"
            >
              -
            </button>
          </td>
        </tr>
        <tr>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              1
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              2
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              3
            </button>
          </td>
          <td>
            <button
              className="operator"
              onClick={[Function]}
              type="button"
            >
              +
            </button>
          </td>
        </tr>
        <tr>
          <td
            colSpan="2"
          >
            <button
              onClick={[Function]}
              type="button"
            >
              0
            </button>
          </td>
          <td>
            <button
              onClick={[Function]}
              type="button"
            >
              .
            </button>
          </td>
          <td>
            <button
              className="operator"
              onClick={[Function]}
              type="button"
            >
              =
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
